File: Employee_system_backend/src/main/java/com/hkdigital/service/EmpService.java
package com.hkdigital.service;

import java.util.List;

import org.springframework.data.repository.CrudRepository;

import com.hkdigital.model.Emp;

public interface EmpService {
	
	
	//update ke liye ye call hoga suppose
	public Emp createEmp(Emp emp);
	
	public List<Emp> getAllEmp();
	
	public Emp getEmpById(int id);
	
	public void deleteEmp(int id);
	
	
	public Emp updateEmp(int id, Emp emp);
	
	
	
	

}

File: react_frontend/src/Service/EmpService.js

import axios from 'axios';


const BASE_API_URL = "http://localhost:8082/api/v1";
class EmpService {

    saveEmp(formData) {
        return axios.post(BASE_API_URL+"/save",formData);
    }

    getAllEmp()
    {
        return axios.get(BASE_API_URL+"/");
    }

    getEmpById(id)
    {
        return axios.get(BASE_API_URL+"/"+id);
    }

    deleteEmp(id){
        return axios.get(BASE_API_URL+"/delete/"+id);
    }

    updateEmp(id,formData)
    {
        return axios.post(BASE_API_URL+"/update/"+id,formData)
    }

}

export default new  EmpService();